dashboard/modules/tune/tune_head.py

```diff
@@ -0,0 +1,233 @@
+        except KeyError:
+            return await rest_response(success=False, message="Bad request")
+
+        err, experiment = self.set_experiment(experiment)
+        if err:
+            return await rest_response(success=False, error=err)
+        return await rest_response(
+            success=True, message="Successfully set experiment", **experiment)
+
+    @routes.get("/tune/enable_tensorboard")
+    async def enable_tensorboard(self, req) -> aiohttp.web.Response:
+        self._enable_tensorboard()
+        if not self._tensor_board_dir:
+            return await rest_response(
+                success=False, message="Error enabling tensorboard")
+        return await rest_response(success=True, message="Enabled tensorboard")
+
+    def get_stats(self):
+        tensor_board_info = {
+            "tensorboard_current": self._logdir == self._tensor_board_dir,
+            "tensorboard_enabled": self._tensor_board_dir != ""
+        }
+        return {
+            "trial_records": copy.deepcopy(self._trial_records),
+            "errors": copy.deepcopy(self._errors),
+            "tensorboard": tensor_board_info
+        }
+
+    def set_experiment(self, experiment):
+        if os.path.isdir(os.path.expanduser(experiment)):
+            self._logdir = os.path.expanduser(experiment)
+            return None, {"experiment": self._logdir}
+        else:
+            return "Not a Valid Directory", None
+
+    def _enable_tensorboard(self):
+        if not self._tensor_board_dir:
+            tb = program.TensorBoard()
+            tb.configure(argv=[None, "--logdir", str(self._logdir)])
+            tb.launch()
+            self._tensor_board_dir = self._logdir
+
+    def _get_availability(self):
+        return {"available": True, "trials_available": self._trials_available}
+
+    def collect_errors(self, df):
+        sub_dirs = os.listdir(self._logdir)
+        trial_names = filter(
+            lambda d: os.path.isdir(os.path.join(self._logdir, d)), sub_dirs)
+        for trial in trial_names:
+            error_path = os.path.join(self._logdir, trial, "error.txt")
+            if os.path.isfile(error_path):
+                self._trials_available = True
+                with open(error_path) as f:
+                    text = f.read()
+                    self._errors[str(trial)] = {
+                        "text": text,
+                        "job_id": os.path.basename(self._logdir),
+                        "trial_id": "No Trial ID"
+                    }
+                    other_data = df[df["logdir"].str.contains(trial)]
+                    if len(other_data) > 0:
+                        trial_id = other_data["trial_id"].values[0]
+                        self._errors[str(trial)]["trial_id"] = str(trial_id)
+                        if str(trial_id) in self._trial_records.keys():
+                            self._trial_records[str(trial_id)]["error"] = text
+                            self._trial_records[str(trial_id)][
```

Comment:
There are too many `str(trail_id)`, can we just `trail_id = str(other_data["trial_id"].values[0])`?